include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,471 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_ANGLE_HPP
+#define SFML_ANGLE_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Represents an angle value
+///
+////////////////////////////////////////////////////////////
```

Comment:
We should mention that the internal representation is radians, and that it should be preferred to degrees whenever possible. Yes, it's a "leaky" detail, but it helps people avoid unnecessary conversions.

Reply:
I'll add a note about that. It's pretty unlikely we ever change the underlying representation so I think this is a reasonable thing to "leak" such a detail. It does call into question whether degrees should be the underlying representation since doing so would likely prevent more lossy floating point operations since most people would rather reason about angles in terms of degrees than radians. We could even do something a little weird and represent angles as millidegrees (or microdegrees?) to avoid the losses of doing many floating point calculations. We don't exactly need the range of `float`. Who really needs angles greater than `std::numeric_limits<sf::Int32>::max()` anyways?

Reply:
>  It does call into question whether degrees should be the underlying representation since doing so would likely prevent more lossy floating point operations since most people would rather reason about angles in terms of degrees than radians. 

That's a good point actually -- and maybe it's a reason why we should _not_ document how it's implemented.

If we document this, then people might specifically optimize for it and make their code more ugly than necessary by using `sf::radians(1.57075f)`. Also, once documented, people have a right to rely on this to remain true throughout SFML 3.

People who are really interested in the internals will just look at the code -- and they can do that, but we give no guarantee. By documenting it, we make it part of the API and encourage a certain way of using `sf::Angle`, which goes against its primary goal.

Reply:
I removed the comment about radians being preferred so we can change our mind in the future. I still want to experiment with something like millidegrees or micro degrees to see if that makes a noticeable improvement in some way.